test/phoenix/router/routing_test.exs

```diff
@@ -77,6 +77,9 @@ defmodule Phoenix.Router.RoutingTest do
 
     conn = call(Router, :get, "/spaced users/hello matey")
     assert conn.params == %{"id" => "hello matey"}
+
+    conn = call(Router, :get, "/users/a+b")
+    assert conn.params == %{"id" => "a b"}
```

Comment:
Do you need to add another test to cover the list format here https://github.com/phoenixframework/phoenix/pull/1749/files#diff-4b61711a8ead48b802b5b21df56705a4R116